Assets/Scripts/HUD/ProgressBar.cs

```diff
@@ -5,39 +5,49 @@
 public class ProgressBar : MonoBehaviour
 {
     private Slider slider;
+    public Gradient gradient;
+    public Image fill;
 
 
     private void Awake()
     {
         slider = gameObject.GetComponent<Slider>();
         slider.maxValue = 100f;
         slider.minValue = 0f;
-        slider.value = 75f;
+        slider.value =5f;
+        fill.color = gradient.Evaluate(0.1f);
     }
 
     private void Update()
     {
         //This should be deleted. Only for testing purposes.
         if (Input.GetKeyDown(KeyCode.Z))
-            DecreasePollution(5f);
+            IncreaseSustainibility(5f);
         if (Input.GetKeyDown(KeyCode.X))
-            IncreasePollution(5f);
+            DecreaseSustainibility(5f);
+       
     }
 
-    public void IncreasePollution (float pollutionChange)
+    public void DecreaseSustainibility (float pollutionChange)
     {
         if (slider.value + pollutionChange > 100f)
             slider.value = slider.maxValue;
         else
             slider.value = (slider.value + pollutionChange);
+
+        fill.color = gradient.Evaluate(slider.normalizedValue);
     }
 
-    public void DecreasePollution (float pollutionChange)
+    public void IncreaseSustainibility (float pollutionChange)
     {
         if(slider.value-pollutionChange < 0)
             slider.value = slider.minValue;
         else
         slider.value = (slider.value - pollutionChange);
+
+        fill.color = gradient.Evaluate(slider.normalizedValue);
```

Comment:
Instead of having those 2 methods which is kinda repeat code, you can have only 1 like this:
![image](https://user-images.githubusercontent.com/54850409/158808753-7408dd40-f84f-4e6f-8c30-a5f9f358261c.png) - This handles both ways, and you dont need to check if the slider value is within the range, as you already pre-set the range before and it should take care of that for you.
Using my method then you can call it like that (if you want to decrease, just add a minus sign):
![image](https://user-images.githubusercontent.com/54850409/158808935-b6ca9aa2-a6cf-4abd-8b31-e72c38955702.png)

Hopefully I explained clearly

Reply:
Thanks! Im on it.